src/dcap_provider.cpp

```diff
@@ -80,20 +80,25 @@ static char PRIMARY_CERT_URL[] =
     "http://169.254.169.254/metadata/THIM/sgx/certification";
 static std::string primary_cert_url = PRIMARY_CERT_URL;
 
+#if defined __SERVICE_VM__
+static char SECONDARY_CERT_URL[] = "http://localhost:4321/tdx/certification";

```

Comment:
Should these names be more specific? For example THIM_AGENT_TDX_CERT_URL? and similar for PRIMARY above?

Reply:
The way we are implementing TDX certificate retrieval is by having a specific build for Service VM, the intention being eliminating any chances of running the SGX path within the service VM (or viceversa) due to an error. This means that we can't have different variables for SGX and TDX.

Primary refers to THIMService url, which isn't used in TDX.

Reply:
Oh ok that makes sense! -- Thanks for clarifying. Would it make sense then to call it THIM_AGENT_CERT_URL and THIM_SERVICE_CERT_URL? -- downstream this feels like it makes more sense than primary/secondary.

Reply:
Yeah, if we ever have the time to refactor DCAP we should do it.

Reply:
Thim agent is an internal term, so primary and secondary makes more sense. It can be updated to anything.